### PR TITLE
Use nbqa-ruff to lint and autofix example notebooks

### PR DESCRIPTION
**Pull request by namurphy**
We currently use [ruff](https://beta.ruff.rs/docs/) as a linter and code quality tool for our Python code.  However, we have not been applying ruff to our example notebooks, so we have not been having the same code quality checks.

This PR enables nbqa-ruff in our suite of pre-commit checks.  This is an [nbqa](https://nbqa.readthedocs.io/en/latest/index.html) tool that runs ruff on our notebooks.

This PR also contains a number of autofixes from applying ruff to our example notebooks.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -372,10 +367,12 @@ max-complexity = 10
 "plasmapy/plasma/sources/*.py" = ["D102"]
 "plasmapy/utils/_pytest_helpers/pytest_helpers.py" = ["BLE001", "PLR", "SLF001"]
 "plasmapy/utils/_pytest_helpers/tests/test_pytest_helpers.py" = ["BLE001", "TRY002"]
+"plasmapy/utils/calculator/plasma_calculator.ipynb" = ["D100", "SLF001"]
```
Comment (namurphy) [inline pyproject.toml:370]:
```suggestion
```

File: pyproject.toml [hunk 2]
```diff
@@ -372,10 +367,12 @@ max-complexity = 10
...
 "plasmapy/utils/decorators/tests/test_converters.py" = ["ARG001", "ARG005"]
 "setup.py" = ["D100"]
 "test_*" = ["D100", "D101", "D102", "D103", "D104", "D209", "D400", "D401", "N801", "SLF001"]
 "*/*/tests/__init__.py" = ["D104"]
+"docs/notebooks/*/*.ipynb" = ["D100", "T201"]
```
Comment (namurphy) [inline pyproject.toml:375]:
```suggestion
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -95,7 +95,8 @@ repos:
 - repo: https://github.com/nbQA-dev/nbQA
   rev: 1.7.0
   hooks:
-  - id: nbqa-black
+  - id: nbqa-check-ast
+  - id: nbqa-ruff
+    args: ['--select=A,ARG,B,BLE,C,C4,E,F,FLY,I002,INT,ISC,PERF,PIE,PLC,PLE,PYI,Q003,RET,RSE,SIM,TID,TRY,UP,W', '--ignore=B007,B008,B015,B018,E402,E501,E741,SIM114']
```
Comment (namurphy) [inline .pre-commit-config.yaml:100]: Some of the ignored rules could be added back in later on, possibly when there are autofixes available.

File: docs/notebooks/getting_started/particles.ipynb [hunk 4]
```diff
@@ -426,7 +426,7 @@
    "outputs": [],
    "source": [
     "argon = Particle(\"Ar 0+\")\n",
-    "argon.ionize(inplace=True)\n",
+    "argon = argon.ionize()\n",
```
Comment (namurphy) [inline docs/notebooks/getting_started/particles.ipynb:429]: I need to revert this, since it's specifically showing off how to use the `inplace` parameter.

**Review by ejohnson-96 — APPROVED** [review]
👍

**Comment by namurphy** [comment]
@ejohnson-96 — thank you for the review!

**Pull request merged**